doc/user/client.md

```diff
@@ -244,6 +244,44 @@ func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, e
 }
 ```
 
+##### Updating Status Subresource
+
+When updating the status subresource from the client, the StatusWriter must be
```

Comment:
Can we link to this document here https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#status-subresource wdyt?

Reply:
Also, might want to highlight in our doc that using `r.Client.Update(ctx, mem)` will NOT update the `status`. I ran across that trying to set status conditions and adding/removing finalizers, which requires two separate updates.

Reply:
Done, thanks guys!